packages/components/timed-text-editor/WrapperBlock.js

```diff
@@ -55,14 +75,38 @@ class WrapperBlock extends React.Component {
       return true;
     }
 
+    if (nextProps.blockProps.isEditable !== this.props.blockProps.isEditable) {
+      return true;
+    }
+
+    if(nextProps.block.getData().get('speaker') !== this.state.speaker){
+      console.log('shouldComponentUpdate wrapper speaker', nextProps.block.getData().get('speaker') , this.state.speaker )
+      return true;
+    }
+
     return false;
   };
 
-  handleOnClickEdit = () => {
-    const newSpeakerName = prompt('New Speaker Name?');
+  componentDidUpdate  = (prevProps, prevState) =>{
+
+    if(prevProps.block.getData().get('speaker') !== prevState.speaker){
+        console.log('componentDidUpdate wrapper speaker', prevProps.block.getData().get('speaker') , prevState.speaker );
+        
+        this.setState({
+          speaker: prevProps.block.getData().get('speaker')
+        })
+
+        return true;
+      }
+  }
 
+  handleOnClickEdit = () => {
+    const oldSpeakerName = this.state.speaker;
+    const newSpeakerName = prompt('New Speaker Name?', this.state.speaker);
     if (newSpeakerName !== '' && newSpeakerName !== null) {
       this.setState({ speaker: newSpeakerName });
+      const isUpdateAllSpeakerInstances = confirm(`Would you like to change all occurrences ${oldSpeakerName} of with ${newSpeakerName} or just this one?\n\n pick OK for all, or CANCEL for just this one.`);
```

Comment:
So playing around with this, I wonder if it's possible to add 3 options rather than OK (all) and CANCEL (one) because it can be misleading, even with the text. Some flashbacks on error prone dialogue windows :D
I would expect something like:

"Would you like to change"
all, one, cancel (default - e.g. when you press enter)

This might mean changing from a single `confirm` to a nested confirm` see (https://stackoverflow.com/a/9091451), because you cannot add more than yes/no to a `confirm`.

Reply:
yeah, fair point, thanks, will have a look into it

Reply:
Ok, I played around with it, I am not sure that the nested prompt would help with clarity. 

I think it might be more a case of better UX writing of the instructions?
eg 
- Speaker name? ok/cancel
- Would you like to apply to this change to all instances? ok/cancel

What do you think?

Reply:
the nested confirm should show two prompts of what you suggested above in succession (speaker name confirm and change all confirm) - and i think the phrasing of what you suggested is good!

Reply:
My bad - you are correct in saying that it is the same set up.
It's mad that the JS cannot be changed (saw [this](https://stackoverflow.com/questions/22885897/javascript-rename-confirm-buttons)... Since I strongly believe in using the safer action as the primary action:
https://uxmovement.com/buttons/visual-weight-of-primary-and-secondary-action-buttons/

I guess it's either you implement your own (which might be better in the longer run, but time consuming), or you just use what they give you. At this stage, until proven that the default button has created problems for the user, I am ok with going with the latter.

Reply:
One feature that would be nice to have with such actions is an UNDO feature.